From 18675a51fca27b324eb4e4b49593bcee9bc4535e Mon Sep 17 00:00:00 2001
From: Jonathan Wakely <jwakely@redhat.com>
Date: Fri, 17 Feb 2017 00:55:25 +0000
Subject: [PATCH] Fix parameter of ImagePyramid::swap

You can't swap with a const object. This fixes a compilation failure
with GCC 7 which rejects the incorrect function even if it isn't
instantiated.
---
 include/vigra/imagecontainer.hxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/vigra/imagecontainer.hxx b/include/vigra/imagecontainer.hxx
index a6f0f80..636cc64 100644
--- a/include/vigra/imagecontainer.hxx
+++ b/include/vigra/imagecontainer.hxx
@@ -763,7 +763,7 @@ public:
         /** swap contents of this array with the contents of other
             (STL-Container interface)
          */
-    void swap(const ImagePyramid<ImageType, Alloc> &other)
+    void swap(ImagePyramid<ImageType, Alloc> &other)
     {
         images_.swap(other.images_);
         std::swap(lowestLevel_, other.lowestLevel_);
